### PR TITLE
Reimplement tf_prefix in Noetic exactly as it was in Melodic

### PR DESCRIPTION
**Pull request by sloretz**
This is an alternative to #139.  The goal of this PR is to make ROS Noetic `robot_state_publisher` behave the same as the ROS Melodic version.  It's different from #139 in these ways:

1. It calls the parameter `tf_prefix`
2. It reimplements the tf1 prefix and frame slash handling exactly as it behaves in ROS Melodic
2. It does not break ABI

While it doesn't break ABI, it effectively breaks using `JointStateListener` with a custom `RobotStatePublisher`. In this PR `JointStateListener` calls non-virtual versions of `publishFixedTransforms()` and `publishTransforms()`. Users who subclass `RobotStatePublisher`, override `publishFixedTransforms()` or `publishTransforms()`, and pass their subclass instance to a vanilla `JointStateListener` will observe their callbacks never being called.

**Note:** ROS 2 `robot_state_publisher` uses a different approach. It has a parameter `frame_prefix` that only appends a string to the frame name - no tf1 style special handling of slashes. See #159 for details.

### CODE REVIEW

File: CMakeLists.txt
Comment (clalancette) [inline CMakeLists.txt:81]: This line is redundant, isn't it?  That is, shouldn't you just be able to add `test/frames_and_slashes.urdf` to the preceding line and be done with it?

**Review by clalancette — APPROVED** [review]
I've left one nit, but otherwise, this is a really good implementation.  One other issue is that CI seems to be failing.  I'm going to approve anyway, assuming that you'll fix both of these issues before merging.

**Comment by jmirabel** [comment]
Any chances this will be merged soon ?

It has been more than a year since the issue has been raised and proposal have been made. I think this functionality is useful and many people would love to use this again.

CI on focal fails because it failed to open the [bag file](http://wiki.ros.org/robot_state_publisher/data?action=AttachFile&do=get&target=joint_states_indexed.bag). Does it ring a bell ?

**Comment by mlanighan** [comment]
Any updates on a timeline for when this will be merged?

File: CMakeLists.txt
Comment (clalancette) [reply CMakeLists.txt:81]: This comment is still open.

Comment (sloretz) [reply CMakeLists.txt:81]: Rebased and removed redundant install in e817dcf

**Review by clalancette — APPROVED** [review]
Looks good to me assuming CI goes green.

**Pull request merged**
